hardening/disa-alignment/oscap/test.py

```diff
@@ -0,0 +1,104 @@
+
+with g.snapshotted():
+    oscap.unselect_rules(util.get_datastream(), 'remediation-ds.xml', remediation.excludes())
+    g.copy_to('remediation-ds.xml')
+
+    for _ in range(2):
+        cmd = [
+            'oscap', 'xccdf', 'eval', '--profile', profile,
+            '--progress', '--remediate', 'remediation-ds.xml',
+        ]
+        proc = g.ssh(' '. join(cmd))
+        if proc.returncode not in [0,2]:
+            raise RuntimeError(f"remediation oscap failed with {proc.returncode}")
+        g.soft_reboot()
+
+    with util.get_content() as content_dir:
+        # There is always one (the latest) DISA benchmark in content src
+        references = content_dir / 'shared' / 'references'
+        disa_ds = next(
+            references.glob(f'disa-stig-rhel{versions.rhel.major}-*-xccdf-scap.xml')
+        )
+
+        # old RHEL-7 oscap mixes errors into --progress rule names without a newline
+        redir = '2>&1' if versions.oscap >= 1.3 else ''
+        # RHEL-7 HTML report doesn't contain OVAL findings by default
+        oval_results = '' if versions.oscap >= 1.3 else '--results results.xml --oval-results'
+
+        shared_cmd = ['oscap', 'xccdf', 'eval', '--progress', oval_results]
+        # Scan with scap-security-guide benchmark
+        g.copy_to(util.get_datastream(), 'ssg-ds.xml')
+        cmd = [
+            *shared_cmd,
+            '--profile', profile,
+            '--report', 'ssg-report.html',
+            '--stig-viewer', 'ssg-stig-viewer.xml',
+            'ssg-ds.xml', redir,
+        ]
+        proc = g.ssh(' '. join(cmd))
+        if proc.returncode not in [0,2]:
+            raise RuntimeError(f"remediation oscap failed with {proc.returncode}")
+        g.copy_from('ssg-report.html')
+        g.copy_from('ssg-stig-viewer.xml')
+
+        # Scan with DISA benchmark
+        g.copy_to(disa_ds, 'disa-ds.xml')
+        cmd = [
+            *shared_cmd,
+            '--profile', '\'(all)\'',
+            '--report', 'disa-report.html',
+            '--results-arf', 'disa-arf.xml',
+            'disa-ds.xml', redir
+        ]
+        proc = g.ssh(' '. join(cmd))
+        if proc.returncode not in [0,2]:
+            raise RuntimeError(f"remediation oscap failed with {proc.returncode}")
+        g.copy_from('disa-report.html')
+        g.copy_from('disa-arf.xml')
+
+        compare_script = content_dir / 'utils' / 'compare_results.py'
+        env = os.environ.copy()
+        env['PYTHONPATH'] = str(content_dir)
+        cmd = [
+            compare_script, 'ssg-stig-viewer.xml', 'disa-arf.xml',
+        ]
+        _, lines = util.subprocess_stream(cmd, env=env)
+
+        # Same result format:      CCE CCI - DISA_RULE_ID SSG_RULE_ID     RESULT
+        # Different result format: CCE CCI - DISA_RULE_ID SSG_RULE_ID     SSG_RESULT - DISA_RESULT
+        result_regex = re.compile(r'[\w-]+ [\w-]+ - [\w-]+ (\w*)\s+(\w+)(?: - *(\w+))*')
+        for match in result_regex.finditer('\n'.join(lines)):
```

Comment:
Okay, that was a bad advice with `re.finditer()`, I didn't realize the only input it has is a full string.

So `util.subprocess_stream()` does all the hard work of efficiently streaming the command output line-by-line, and that gets all thrown away by `.join()` just reading everything, merging it into one giant string, passing it to `.finditer()` to be broken down again and iterated over.

If the output is not too huge (ie. not Anaconda output), this might be okay, but the previous solution is IMHO better. Your call on which one you want to use.

Reply:
If you really want `re.finditer()`, then I would just do
```python
proc = util.subprocess_run(cmd, env=env)
```
above, and then just pass `proc.stdout.rstrip('\n')` to `re.finditer()` instead of `.join()`ing the lines together.